pkg/apis/core/validation/validation.go

```diff
@@ -2481,10 +2481,10 @@ func validatePersistentVolumeClaimResourceKey(value string, fldPath *field.Path)
 }
 
 var resizeStatusSet = sets.New(core.PersistentVolumeClaimControllerResizeInProgress,
-	core.PersistentVolumeClaimControllerResizeFailed,
+	core.PersistentVolumeClaimControllerResizeInfeasible,
```

Comment:
We may want to allow updates of existing objects with the old status, that way somebody can still delete the object. Blocking new objects and blocking updates to the old status values seems reasonable.

Reply:
So, during status validation - we do not actually check for status values on old PVC and hence it is possible for existing object's status to be updated to *valid* values.  Having said that - if *other* fields of PVC is being updated and it contains renamed values, then those *other* fields can't be updated, until PVC is fixed with new values for `allocatedResourceStatus` field. 

But - if we allow PVC with invalid `allocatedResourceStatus` to have other update operations, then we need to carry those "invalid" types somewhere in the code. It will be tricky to do this correctly.

Reply:
The old condition is alpha, I think we _can_ drop it without replacement.

If we choose to change the conditions on PVC update, how long will we need to carry that code? It will be triggered by PVC update in the API server. If I quickly upgrade my API server from 1.30 to 1.31 and 1.32, without changing any PVC, we still need the code in 1.32 and who knows what newer release.